### PR TITLE
Update the COC and TOS Description on the Creator Settings Form

### PR DESCRIPTION
**Pull request by juliannatetreault**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
This PR refactors the COC and TOS description on the Creator Settings form (per @anujbhavsar96's design feedback) by doing the following:
- [x] Moving the COC and TOS description text below the checkbox legend
- [x] Un-italicizing the text and instead styling the text with the `crayons-field__description` class

## Related Tickets & Documents
Relates to issue #15524 

#### Please note that you will need to essentially clear out your database to take this feature for a spin. 🏎️  

To do so, you can either drop your entire database via `rails:db drop` 

_**or**_

You can open a rails console via `rails c` and do the following:
 ```
# Enable the feature flag
FeatureFlag.enable(:creator_onboarding)

# Delete all users 
User.destroy_all

# Set `waiting_on_first_user` to be true
Settings::General.waiting_on_first_user = true

# If you've logged in before set the mascot id to nil
 Settings::General.mascot_user_id = nil
```

Now, run your server and navigate to `http://localhost:3000/` and complete the "Create Account" form. After completing the "Create Account" form, you should be brought to the new Creator Setup page, `${baseUrl}/admin/creator_settings/new?referrer=${baseUrl}`.

#### Before:
Screen Shot 2021-11-19 at 9 44 21 AM

#### After:
Screen Shot 2021-11-29 at 2 07 28 PM

### UI accessibility concerns?
I should have addressed any UI accessibility concerns in this PR!

## Added/updated tests?

- [ ] Yes
- [x] No, and this is why: there is already an E2E test that tests this.
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://developers.forem.com) or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] This PR changes the Forem platform and our documentation needs to be
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [x] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?
N/A

## [optional] What gif best describes this PR or how it makes you feel?

Sandy Cheeks adjusting tubes attached to SpongeBob

### CODE REVIEW

**Review by Zhao-Andy — APPROVED** [review]

File: app/views/admin/creator_settings/_form.html.erb [hunk 1]
```diff
@@ -83,9 +83,10 @@
 </div>
 
 <div class="crayons-field mt-6 align-left">
-  <fieldset aria-describedby="section-description">
+  <fieldset aria-describedby="terms-description">
```
Comment (jeremyf) [inline app/views/admin/creator_settings/_form.html.erb:86]: I'm a bit confused by the change to `aria-describedby` (and the original value).  I'm looking at the DOM and I don't see an element with the ID `terms-description`.  That's not a blocker for this change, but perhaps an accessibility issue.

File: app/views/admin/creator_settings/_form.html.erb [hunk 2]
```diff
@@ -83,9 +83,10 @@
...
     <legend class="crayons-field__label mb-2">Finally, please agree to the following:</legend>
     <div>
+    <p id="terms-subtitle" class="crayons-field__description">You will have the opportunity to establish the Code of Conduct and Terms and Conditions for the users of your Forem during the setup process.</p>
```
Comment (jeremyf) [inline app/views/admin/creator_settings/_form.html.erb:89]: In reading the text, I was assuming that immediately following "Finally, please agree to the following:" would be a bulleted list.  Would it make sense to move the "You will have the opportunity…" to after the bulleted list.  Or before?

**Review by anujbhavsar96 — COMMENTED** [review]
This looks perfect✨. Just some nitpicking from my end. Don't hate me for this @juliannatetreault 🙃

I feel the spacing between label and description is a bit higher. And spacing between description and checkbox+contennt is lower. 

Those green marked spacing will look fine.
Group 1

File: app/views/admin/creator_settings/_form.html.erb (hunk 1 above)
Comment (juliannatetreault) [reply app/views/admin/creator_settings/_form.html.erb:86]: Good catch. I've reverted this change, @jeremyf!

**Comment by juliannatetreault** [comment]
> This looks perfect✨. Just some nitpicking from my end. Don't hate me for this @juliannatetreault 🙃
> 
> I feel the spacing between label and description is a bit higher. And spacing between description and checkbox+contennt is lower.
> 
> Those green marked spacing will look fine. Group 1

Thank you so much for the feedback, @anujbhavsar96! I've updated the spacing per your request. 😊

**Review by joshpuetz — APPROVED** [review]

**Pull request merged**